source/common/config/type_to_endpoint.cc

```diff
@@ -84,6 +84,23 @@ const Protobuf::MethodDescriptor& restMethod(absl::string_view type_url) {
   return *Protobuf::DescriptorPool::generated_pool()->FindMethodByName(method_name);
 }
 #undef API_TYPE_URL_IS
+} // namespace
+
+const Protobuf::MethodDescriptor&
+xdsCarrierMethod(absl::string_view type_url,
+                 envoy::api::v2::core::ApiConfigSource::ApiType api_type) {
+  switch (api_type) {
+  case envoy::api::v2::core::ApiConfigSource::REST:
+    return restMethod(type_url);
+  case envoy::api::v2::core::ApiConfigSource::GRPC:
+    return sotwGrpcMethod(type_url);
+  case envoy::api::v2::core::ApiConfigSource::DELTA_GRPC:
+    return deltaGrpcMethod(type_url);
+  default:
+    return *Protobuf::DescriptorPool::generated_pool()->FindMethodByName(
```

Comment:
you should probably assert here instead

Reply:
+1 this should be NOT_REACHED due to schema checking.

Reply:
Before sending out the PR I had tried the GCOVR NOT_REACHED thing, and found that there is at least one test (for UNSUPPORTED_REST_LEGACY) that does reach here. Until UNSUPPORTED_REST_LEGACY is more definitively removed, with its throw replaced by a NOT_REACHED, I think the cleanest way possible is to just let FindMethodByName return whatever "not found" descriptor.

Reply:
...ah, but, there's apparently a potentially uninitialized value that msan/asan are complaining about. I slightly suspect it's what we're talking about here. Not really sure what else it could be.

Reply:
Then shouldn't we test for UNSUPPORTED_REST_LEGACY and throw on that directly?

Reply:
There is a test for that. I meant that where the method_descriptor is currently gotten, it will be gotten for UNSUPPORTED_REST_LEGACY as well. That's unavoidable, other than repeating the function call for each switch case. Maybe that would be best, despite being ugly?

However, I just tried reverting all of the changes and running my local clang-msan again, and I still got the same crash. clang-asan doesn't work for me and my machine apparently doesn't have enough disk space to run the bazel.asan CI script. So... I don't have a way to actually engage with this problem. Since (presumably) just dropping all of the changes other than the integration test should avoid the problem, I'm thinking maybe it's best to drop these changes. They're kind of just a nice-to-have cleanup anyways; the integration test is the real safety measure.

Reply:
OK I don't really have enough context. As long as the integration test catches the original bug it's fine.

Reply:
Done.